src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  static newAnimal(animalObj) {
+    this.alive.push(animalObj);
+  }
+
+  static animalDeath(animalObj) {
+    const animalIndex = this.alive.indexOf(animalObj);
+
+    this.alive.splice(animalIndex, 1);
+  }
+
+  constructor(name, health) {
+    const defaultHealth = 100;
+
+    this.name = name;
+    this.health = health || defaultHealth;
+
+    Animal.newAnimal(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    const bitePower = 50;
+    const isHerbivore = Object.hasOwnProperty.call(victim, 'hidden');
```

Comment:
there's a special method for checking what instance our object is. Checking for some exact property is a crutch.
What if we add the `hidden` property to the Carnivores?